plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -57,22 +143,44 @@ const productGallery = {
 				context.imageIds.length
 			);
 
-			context.selectedImageNumber = newImageNumber;
+			actions.userHasInteracted();
```

Comment:
Please rebase with `trunk` so this PR is included: https://github.com/woocommerce/woocommerce/pull/55976. At the moment changing images using keyboard arrows doesn't load the images.

https://github.com/user-attachments/assets/8e9b5160-a4cc-4a44-8432-1ebbcd2d40ed